---
content_type: page
title: Lecture Notes
uid: 7d35d7d9-97e3-ac34-cd3a-945090b26e21
---

{{< tableopen >}}
{{< theadopen >}}
{{< tropen >}}
{{< thopen >}}
LEC #
{{< thclose >}}
{{< thopen >}}
TOPICS
{{< thclose >}}

{{< trclose >}}

{{< theadclose >}}
{{< tropen >}}
{{< tdopen colspan="3" >}}
**Introduction to Biology**
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
1
{{< tdclose >}}
{{< tdopen >}}
The Central Dogma: Some Algorithms Introduction
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen colspan="3" >}}
**Enumerative Solutions: Partial Digest Problem and Median Strings**
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
2
{{< tdclose >}}
{{< tdopen >}}
Partial Digest Problem
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
3
{{< tdclose >}}
{{< tdopen >}}
Motifs and Median Strings ([PDF]({{< baseurl >}}/resources/lecture_03)) (Courtesy of Jerome Mettetal. Used with permission.)  
  
A Review Paper on Regulatory Motifs
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen colspan="3" >}}
**Dynamic Programming: Sequence Alignments**
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
4
{{< tdclose >}}
{{< tdopen >}}
Global Alignment ([PDF]({{< baseurl >}}/resources/lecture_04)) (Courtesy of Russ Cox and Michael Walfish. Used with permission.)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
5
{{< tdclose >}}
{{< tdopen >}}
Local Alignment ([PDF]({{< baseurl >}}/resources/lecture_05)) (Courtesy of Anthony W. Scelfo and Athicha Muthitacharoen. Used with permission.)  
  
[More on Alignment Statistics](http://www.pubmedcentral.nih.gov/articlerender.fcgi?tool=pubmed&pubmedid=11139604)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
6
{{< tdclose >}}
{{< tdopen >}}
Spliced Alignment ([PDF]({{< baseurl >}}/resources/lecture_06)) (Courtesy of Lele Yu and Sam Kaufman. Used with permission.)  
  
[A Paper on the Subtle Problem of Aligning Proteins to DNA](http://portal.acm.org/citation.cfm?id=267521.267893)  
  
[Spliced Alignment with SIM4](http://www.genome.org/cgi/content/full/8/9/967)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
7
{{< tdclose >}}
{{< tdopen >}}
More Efficient Alignment ([PDF]({{< baseurl >}}/resources/lecture_07)) (Courtesy of Mark Halsey and Robert Edward Beverly. Used with permission.)  
  
A Recent Paper on Sub-quadratic Time Alignment: Crochemore, Maxime, Gad M. Landau, and Michal Ziv-Ukelson. "A Subquadratic Sequence Alignment Algorithm for Unrestricted Scoring Matrices." _SIAM J. COMPUT._ 32, no. 6: 1654-1673.  
  
Gene Myers on the State of Alignment in 1991: Myers, E. W. _An overview of sequence comparison algorithms in molecular biology._ University of Arizona, Department of Computer Science, Technical Report TR 91-29, 1991.
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen colspan="3" >}}
**Graph Theory: Sequencing Genes and Proteins**
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
8
{{< tdclose >}}
{{< tdopen >}}
Genomics and SBH Graphs  
  
A Nice Paper on Getting SBH to Work: Lengauer, Thomas, David Sankoff, Sorin Istrail, Pavel Peuvzner, and Michael Waterman. "Annual Conference on Research in Computational Molecular Biology Proceedings of the fifth annual international conference on Computational biology Montreal, Quebec, Canada." New York: ACM Press, 2001, pp. 269-277. ISBN: 1581133537.
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
9
{{< tdclose >}}
{{< tdopen >}}
Peptide Graphs ([PDF]({{< baseurl >}}/resources/lecture_09)) (Courtesy of Robert Edward Beverly and Christopher J. Peikert. Used with permission.)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen colspan="3" >}}
**Pattern Matching: Exact Matches, Gapless Alignments, and BLAST**
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
10
{{< tdclose >}}
{{< tdopen >}}
Exact Pattern Matching ([PDF]({{< baseurl >}}/resources/lecture_10)) (Courtesy of Lele Yu and Mark Halsey. Used with permission.)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
11
{{< tdclose >}}
{{< tdopen >}}
Suffix Trees ([PDF]({{< baseurl >}}/resources/lecture_11)) (Courtesy of Athicha Muthitacharoen and Anthony W. Scelfo. Used with permission.)  
  
[Early Application of Suffix Trees](http://nar.oupjournals.org/cgi/content/full/27/11/2369)  
  
A Review of Suffix Trees ([PDF](http://citeseer.ist.psu.edu/cache/papers/cs/18442/http:zSzzSzwww.info.uniroma2.itzSz~italianozSzPaperszSzsuffix.pdf/grossi93suffix.pdf))
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
12
{{< tdclose >}}
{{< tdopen >}}
Suffix Arrays and BWTs ([PDF]({{< baseurl >}}/resources/lecture_12)) (Courtesy of Tushara C. Karunaratna and Peter Wai Kei Lee. Used with permission.)  
  
[A Nice Paper on CSAs/BWTs](http://www.ncbi.nlm.nih.gov/pubmed/11791236)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
13
{{< tdclose >}}
{{< tdopen >}}
BLAST ([PDF]({{< baseurl >}}/resources/lecture_13)) (Courtesy of Eitan Reich and Peter Lee. Used with permission.)  
  
[The BLAST Paper](http://www.sciencedirect.com/)  
  
BLAST Statistics and PAMs: F., Altschul S. "Amino acid substitution matrices from an information theoretic perspective." _J Mol Biol_. 219, no. 3 (1991): 555-65.  
  
Introduction of Gapped Seed Filtration
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen colspan="3" >}}
**Clustering: Microarrays and Phylogeny**
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
14
{{< tdclose >}}
{{< tdopen >}}
Clustering (Guest Lecturer)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
15
{{< tdclose >}}
{{< tdopen >}}
Trees ([PDF]({{< baseurl >}}/resources/lecture_15)) (Courtesy of Eugenia Lyashenko. Used with permission.)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen colspan="4" >}}
**[Neighbor Joining](https://academic.oup.com/mbe/article/4/4/406/1029664)**
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
16
{{< tdclose >}}
{{< tdopen colspan="2" >}}
[A Nice Review of Phylogenetic Analysis](http://arjournals.annualreviews.org/doi/pdf/10.1146/annurev.genet.30.1.371?cookieSet=1)  
  
A Very Complicated Paper on Phylogeny: K. Rice, and T. Warnow. "Parsimony is Hard to Beat, Source Lecture Notes." In _Computer Science_ 1276 (1997): 124-133. Archive Proceedings of the Third Annual International Conference on Computing and Combinatorics table of contents. ISBN: 354063357X.  
  
[A Review of Coalescent Theory in Biology](http://www.nature.com/cgi-taf/)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
17
{{< tdclose >}}
{{< tdopen >}}
Application: Microarrays (Guest Lecturer)  
  
The Classic AML/ALL Cancer Classification Paper: Golub, T. R., D. K. Slonim, P. Tamayo, C. Huard, M. Gaasenbeek, J. P. Mesirov, H. Coller, M. L. Loh, J. R. Downing, M. A. Caligiuri, C. D. Bloomfield, and E. S. Lander. "Molecular Classification of Cancer: class Discovery and Class Prediction by Gene Expression Monitoring." _Science_ 286 (15 October 1999).  
  
Finding Weak Signals in Microarray Data ([PDF](http://www.nature.com/cgi-taf/DynaPage.taf?file=/ng/journal/v34/n3/full/ng1180.html&filetype=pdf))  
  
[A Review Article on Mining Microarray Data](http://www.nature.com/cgi-taf/DynaPage.taf?file=/ng/journal/v32/n4s/full/ng1033.html)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen colspan="4" >}}
**Probabilistic Models and Machine Learning: Gene Annotation and Evolution**
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
18
{{< tdclose >}}
{{< tdopen >}}
Hidden Markov Models I ([PDF]({{< baseurl >}}/resources/lecture_17)) (Courtesy of Han-Pang Chiu and Russ Cox. Used with permission.)  
  
An HMM Paper, focused on Recognizing Folds: Krogh, Anders, Michael Brown, I. Saira Mian, Kimmen Sjölander, and David Haussler. "Hidden Markov Models in Computational Biology: Applications to Protein Modeling." _Journal of Molecular Biology_ 235, no. 5 (Feb 3, 1994): 1501-1531.
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
19
{{< tdclose >}}
{{< tdopen >}}
Hidden Markov Models II ([PDF]({{< baseurl >}}/resources/lecture_18)) (Courtesy of Chris Peikert. Used with permission.)  
  
One of the GENSCAN Papers: Burge, Chris, and Samuel Karlin. "Prediction of Complete Gene Structures in Human Genomic DNA." _J. Mol. Biol._ 268 (1997): 78-94.  
  
[An HMM Searching for Coding RNA](http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=64605&rendertype=abstract)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
20
{{< tdclose >}}
{{< tdopen >}}
Gibbs Sampling ([PDF]({{< baseurl >}}/resources/lecture_19)) (Courtesy of Tushara C. Karunaratna. Used with permission.)  
  
[A Seminal Paper on Gibbs Sampling](http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?holding=npg&cmd=Retrieve&db=PubMed&list_uids=8211139&dopt=Abstract)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
21
{{< tdclose >}}
{{< tdopen colspan="2" >}}
Random Projections ([PDF]({{< baseurl >}}/resources/lecture_20)) (Courtesy of Eugenia Lyashenko. Used with permission.)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
22
{{< tdclose >}}
{{< tdopen >}}
MCMC and Bayesian Networks  
Another Probabilistic Method to Phase Haplotype Data
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen colspan="4" >}}
**Horizons**
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
23
{{< tdclose >}}
{{< tdopen >}}
The Future: Protein Structure (Guest Lecturer)
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
24
{{< tdclose >}}
{{< tdopen >}}
The Future: Haplotype Mapping (Guest Lecturer)  
  
Ancestor Reconstruction by Haplotype Coloring: Schwartz, R., and A. Clark, S. Istrail. "Methods for inferring block-wise ancestral history from haploid sequences: The haplotype coloring problem." Proceedings of the 2nd Workshop on Algorithms in Bioinformatics, (WABI02), Springer Lecture Notes in Computer Science (2002): 44-59.
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
25
{{< tdclose >}}
{{< tdopen >}}
Presentations of Final Projects
{{< tdclose >}}

{{< trclose >}}
{{< tropen >}}
{{< tdopen >}}
26
{{< tdclose >}}
{{< tdopen >}}
Presentations of Final Projects (cont.)
{{< tdclose >}}

{{< trclose >}}

{{< tableclose >}}